src/tests/ftest/control/daos_snapshot.py

```diff
@@ -0,0 +1,154 @@
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+'''
+from apricot import TestWithServers, skipForTicket
+from daos_utils import DaosCommand
+from test_utils_container import TestContainer
+
+
+class DaosSnapshotTest(TestWithServers):
+    """
+    Test Class Description:
+        Test daos container create-snap, list-snaps, and destroy-snap. Test
+        steps:
+        1. Create 5 snapshots. Obtain the epoch value each time we create a
+            snapshot.
+        2. List the 5 snapshots, obtain the epoch values, and compare against
+            those returned during create-snap.
+        3. Destroy all the snapshots one by one.
+        4. List and verify that there's no snapshot.
+
+        Test destroy-snap --epcrange.
+        1. Create 5 snapshots.
+        2. Destory all the snapshots with --epcrange. Use the first epoch for B
+            and the last epoch for E.
+        3. List and verify that there's no snapshot.
+
+    Note that we keep the test steps basic due to time constraint. Add more
+    cases if we see bugs around this feature.
+
+    :avocado: recursive
+    """
+    def __init__(self, *args, **kwargs):
+        """Initialize a DaosSnapshotTest object."""
+        super(DaosSnapshotTest, self).__init__(*args, **kwargs)
+        self.daos_cmd = None
+        self.svc = None
+
+    def create_snapshot(self, pool_uuid, svc, cont_uuid, count):
+        """Create snapshots and return the epoch values obtained from stdout.
+
+        Args:
+            pool_uuid (String): Pool UUID.
+            svc (String): Service replicas.
+            cont_uuid (String): Container UUID.
+            count (Integer): Number of snapshots to create.
+
+        Returns:
+            List of string: Epochs obtained from stdout.
+        """
+        epochs = []
+        for _ in range(count):
+            epochs.append(
+                self.daos_cmd.container_create_snap(pool=pool_uuid,
+                cont=cont_uuid, svc=svc)["epoch"])
+        return epochs
+
+    def prepare_pool_container(self):
+        """Create a pool and a container and prepare for the test cases.
+        """
+        self.daos_cmd = DaosCommand(self.bin)
+        self.add_pool(connect=False)
+        self.svc = ",".join(str(rank) for rank in self.pool.svc_ranks)
+        self.container = TestContainer(
+            pool=self.pool, daos_command=self.daos_cmd)
+        self.container.get_params(self)
+        self.container.create()
```

Comment:
FYI: If either https://github.com/daos-stack/daos/pull/3318 or https://github.com/daos-stack/daos/pull/3318 lands before this PR there will be the option to simplify this, as the `TestWithServers.get_container()` method will support defining its `TestContainer` with a `DaosCommand` object.  This enables creating the container with the `daos` command via the `TestWithServers.add_container()` method if the test yaml specifies `control_method: daos`:
```suggestion
        self.add_container(self.pool)
```